test/honest_test.go

```diff
@@ -2,254 +2,170 @@ package test
+			participantCounts: blsParticipantCount,
 		},
 	}
 	for _, test := range tests {
 		test := test
-		t.Run(test.name, func(t *testing.T) {
-			tsg := sim.NewTipSetGenerator(tipSetGeneratorSeed)
-			baseChain := generateECChain(t, tsg)
-			oneChain := baseChain.Extend(tsg.Sample())
-			anotherChain := baseChain.Extend(tsg.Sample())
-			test.options = append(test.options,
-				sim.WithBaseChain(&baseChain),
-				sim.AddHonestParticipants(1, sim.NewFixedECChainGenerator(oneChain), uniformOneStoragePower),
-				sim.AddHonestParticipants(1, sim.NewFixedECChainGenerator(anotherChain), uniformOneStoragePower),
-			)
-			sm, err := sim.NewSimulation(test.options...)
-			require.NoError(t, err)
-			require.NoErrorf(t, sm.Run(1, maxRounds), "%s", sm.Describe())
-			requireConsensusAtFirstInstance(t, sm, baseChain...)
-		})
-	}
-}
-
-func TestSync_AgreementWithRepetition(t *testing.T) {
-	repeatInParallel(t, 50, func(t *testing.T, repetition int) {
-		honestCount := 3 + repetition
-		tsg := sim.NewTipSetGenerator(tipSetGeneratorSeed)
-		baseChain := generateECChain(t, tsg)
-		someChain := baseChain.Extend(tsg.Sample())
-		sm, err := sim.NewSimulation(syncOptions(
-			sim.WithBaseChain(&baseChain),
-			sim.AddHonestParticipants(honestCount, sim.NewFixedECChainGenerator(someChain), uniformOneStoragePower),
-		)...)
-		require.NoError(t, err)
-
-		require.NoErrorf(t, sm.Run(1, maxRounds), "%s", sm.Describe())
-		// Synchronous, agreeing groups always decide the candidate.
-		requireConsensusAtFirstInstance(t, sm, someChain.Head())
-	})
-}
-
-func TestAsyncAgreement(t *testing.T) {
-	if testing.Short() {
-		t.Skip("skipping test in short mode.")
-	}
-
-	t.Parallel()
-	// These iterations are much slower, so we can't test as many participants.
-	for n := 3; n <= 16; n++ {
-		honestCount := n
-		t.Run(fmt.Sprintf("honest count %d", honestCount), func(t *testing.T) {
-			repeatInParallel(t, asyncIterations, func(t *testing.T, repetition int) {
-				tsg := sim.NewTipSetGenerator(tipSetGeneratorSeed)
-				baseChain := generateECChain(t, tsg)
-				someChain := baseChain.Extend(tsg.Sample())
-				sm, err := sim.NewSimulation(asyncOptions(t, repetition,
+		for _, participantCount := range test.participantCounts {
+			participantCount := participantCount
+			name := fmt.Sprintf("%s %d", test.name, participantCount)
+			t.Run(name, func(t *testing.T) {
+				test.options = append(test.options,
 					sim.WithBaseChain(&baseChain),
-					sim.AddHonestParticipants(
-						honestCount,
-						sim.NewFixedECChainGenerator(someChain),
-						uniformOneStoragePower),
-				)...)
+					sim.AddHonestParticipants(participantCount, sim.NewFixedECChainGenerator(targetChain), uniformOneStoragePower))
+				sm, err := sim.NewSimulation(test.options...)
 				require.NoError(t, err)
 				require.NoErrorf(t, sm.Run(1, maxRounds), "%s", sm.Describe())
-				requireConsensusAtFirstInstance(t, sm, baseChain.Head(), someChain.Head())
```

Comment:
The current test failure is due to this. When async, the nodes may not decide the target chain, and instead decide basechain. When sync, they will always decide the target. It's possible the TestHonest_ChainAgreement "async pair" above was just getting lucky with the random seed and managing to decide target, because I'm not sure it must.

Anyway, when async the baseChain must be a valid decision. But when sync we can and should make a stronger assertion of progress.